authorize/filter.go

```diff
@@ -1,44 +1,75 @@
 package authorize
 
-// Condition is just bool condition for whitelisting token perms.
-type Condition func(tokenPerms []string) bool
+import (
+	"fmt"
+	"strings"
+)
+
+// Condition is used to check whether user with tokenPerms has access.
+type Condition struct {
+	isSatisfiedBy func(tokenPerms []string) bool
+	stringRepr    string
+}
 
 // OR is an array of conditions with logic OR. If no condition is passed it returns false.
 func OR(conditions ...Condition) Condition {
-	return func(tokenPerms []string) bool {
-		for _, condition := range conditions {
-			if condition(tokenPerms) {
-				return true
+	return Condition{
+		isSatisfiedBy: func(tokenPerms []string) bool {
+			for _, condition := range conditions {
+				if condition.isSatisfiedBy(tokenPerms) {
+					return true
+				}
 			}
-		}
-		return false
+			return false
+		},
+		stringRepr: mkStringRepr(conditions, " || "),
 	}
 }
 
 // AND is an array of conditions with logic AND. If no condition is passed it returns false.
 func AND(conditions ...Condition) Condition {
-	return func(tokenPerms []string) bool {
-		if len(conditions) == 0 {
-			return false
-		}
-
-		for _, condition := range conditions {
-			if !condition(tokenPerms) {
+	return Condition{
+		isSatisfiedBy: func(tokenPerms []string) bool {
+			if len(conditions) == 0 {
```

Comment:
Btw this is mathematically incorrect. `forall(x: f(x))` is always true if x is empty set. `exists(x: f(x))` is on the other hand false for empty set because there is no x in empty set.

Reply:
yea, `And()` seems like an error, not actual possible invocation. Maybe we should return some `errCondition` or return just error in this case?